api/src/main/java/org/openmrs/module/idgen/IdgenUtil.java

```diff
@@ -98,4 +101,32 @@ public static long convertFromBase(String s, char[] baseCharacters) {
 		}
 		return contents;
 	}
+	
+	/**
+	 * Convenience method for getting a valid prefix from a parent location with a valid prefix attribute up the tree
+	 * 
+	 * @param location the child location. Normally the location for the current {@link UserContext}
```

Comment:
>`"The starting point location"`

Don't assume who's calling this method, you made it `public static` so it's reusable at will, it might not be used with `UserContext`.

Reply:
What do you mean by:
>it might not be used in conjunction with UserContext

Reply:
No no, I have understood what you meant. Makes sense